src/Hocon.Configuration/Config.cs

```diff
@@ -18,82 +20,63 @@ namespace Hocon
 
-        static Config()
+        public static Config Create(HoconElement root)
         {
-            EmptyValue = new HoconValue(null);
-            EmptyValue.Add(new HoconObject(EmptyValue));
+            return new Config(root);
         }
 
-        [Obsolete("For json serialization/deserialization only", true)]
-        protected Config()
+        private Config()
         {
+            Root = this;
         }
 
         /// <inheritdoc />
         /// <summary>
         ///     Initializes a new instance of the <see cref="Config" /> class.
         /// </summary>
-        protected Config(HoconValue value)
-        {
-            Value = (HoconValue)value.Clone(null);
-            Root = (HoconValue)value.Clone(null);
-        }
-
-        /// <inheritdoc cref="Config(HoconValue)" />
-        protected Config(HoconValue value, Config fallback) : this(value)
-        {
-            MergeConfig(fallback);
-        }
-
-        /// <inheritdoc cref="Config(HoconValue)" />
-        /// <param name="root">The root node to base this configuration.</param>
-        /// <exception cref="T:System.ArgumentNullException">"The root value cannot be null."</exception>
-        public Config(HoconRoot root)
+        public Config(HoconElement root):base(root)
         {
-            Value = (HoconValue)root.Value.Clone(null);
-            Root = (HoconValue)root.Value.Clone(null);
-
-            if (!(root is Config cfg))
-                return;
-
-            foreach(var value in cfg._fallbacks)
+            if(root is Config cfg)
+            {
+                Root = cfg.Root;
+                _fallbacks = cfg._fallbacks.ToList();
+            }  
+            else 
             {
-                InsertFallbackValue(value);
+                Root = this;
             }
         }
 
-        /// <inheritdoc cref="Config(HoconValue)" />
-        /// <param name="root">The configuration to use as the primary source.</param>
-        /// <param name="fallback">The configuration to use as a secondary source.</param>
-        /// <exception cref="ArgumentNullException">The source configuration cannot be null.</exception>
-        public Config(HoconRoot root, Config fallback) : this(root)
-        {
-            MergeConfig(fallback);
-        }
-
-        /// <summary>
-        ///     Identical to <see cref="HoconConfigurationFactory.Empty" />.
-        /// </summary>
-        /// <remarks>
-        ///     Added for brevity and API backwards-compatibility with Akka.Hocon.
-        /// </remarks>
-        public static Config Empty => CreateEmpty();
-
-        protected List<HoconValue> _fallbacks { get; } = new List<HoconValue>();
-        public virtual IReadOnlyList<HoconValue> Fallbacks => _fallbacks.ToList().AsReadOnly();
+        protected List<HoconObject> _fallbacks { get; } = new List<HoconObject>();
+        public virtual IReadOnlyList<HoconObject> Fallbacks => _fallbacks.ToList().AsReadOnly();
```

Comment:
No need to call `ToList().AsReadOnly()` - `List<T>` is castable to `IReadOnlyList<T>`

Reply:
Defensive coding, I want to be 100% sure that the user could not change the internal representation of the fallback list.